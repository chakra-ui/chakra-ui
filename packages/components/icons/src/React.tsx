import { createIcon } from "@chakra-ui/icon"

export const ReactIcon = createIcon({
  displayName: "ReactIcon",
  path: (
    <path
      d="M12 13.5C11.6022 13.5 11.2206 13.342 10.9393 13.0607C10.658 12.7793 10.5 12.3978 10.5 12C10.5 11.6022 10.658 11.2206 10.9393 10.9393C11.2206 10.658 11.6022 10.5 12 10.5C12.3978 10.5 12.7794 10.658 13.0607 10.9393C13.342 11.2206 13.5 11.6022 13.5 12C13.5 12.3978 13.342 12.7793 13.0607 13.0607C12.7794 13.342 12.3978 13.5 12 13.5ZM11.472 16.494C11.647 16.704 11.823 16.908 12 17.103C12.177 16.908 12.353 16.705 12.528 16.494C12.176 16.5015 11.824 16.5015 11.472 16.494ZM9.477 16.369C8.70906 16.2891 7.9462 16.1662 7.192 16.001C7.117 16.351 7.06 16.691 7.022 17.017C6.832 18.6 7.097 19.562 7.5 19.794C7.903 20.027 8.868 19.775 10.145 18.82C10.408 18.623 10.673 18.404 10.939 18.165C10.4185 17.5941 9.9304 16.9945 9.477 16.369ZM16.808 16.001C16.091 16.161 15.325 16.285 14.523 16.369C14.0696 16.9945 13.5815 17.5941 13.061 18.165C13.327 18.405 13.592 18.623 13.855 18.82C15.132 19.775 16.097 20.027 16.5 19.794C16.903 19.562 17.167 18.6 16.979 17.017C16.9377 16.6759 16.881 16.3369 16.809 16.001H16.808ZM18.258 15.614C18.835 18.253 18.532 20.354 17.25 21.094C15.968 21.834 13.997 21.046 12 19.227C10.003 21.046 8.032 21.833 6.75 21.093C5.468 20.353 5.165 18.253 5.741 15.613C3.167 14.794 1.5 13.48 1.5 12C1.5 10.52 3.167 9.20699 5.741 8.38599C5.165 5.74699 5.468 3.64599 6.75 2.90599C8.032 2.16599 10.003 2.95399 12 4.77299C13.997 2.95399 15.968 2.16699 17.25 2.90699C18.532 3.64699 18.835 5.74699 18.259 8.38699C20.833 9.20599 22.5 10.52 22.5 12C22.5 13.48 20.833 14.793 18.259 15.614H18.258ZM10.938 5.83499C10.6837 5.6048 10.4191 5.38624 10.145 5.17999C8.868 4.22499 7.903 3.97299 7.5 4.20599C7.097 4.43799 6.833 5.39999 7.021 6.98299C7.061 7.30999 7.117 7.64899 7.191 7.99899C7.94553 7.83373 8.70872 7.71087 9.477 7.63099C9.952 6.97799 10.442 6.37699 10.939 5.83499H10.938ZM14.523 7.63099C15.325 7.71499 16.091 7.83999 16.808 7.99899C16.883 7.64899 16.94 7.30899 16.978 6.98299C17.168 5.39999 16.903 4.43799 16.5 4.20599C16.097 3.97299 15.132 4.22499 13.855 5.17999C13.5806 5.38622 13.3156 5.60477 13.061 5.83499C13.558 6.37699 14.048 6.97799 14.523 7.63099ZM12.528 7.50599C12.353 7.29599 12.177 7.09199 12 6.89699C11.823 7.09199 11.647 7.29499 11.472 7.50599C11.824 7.49852 12.176 7.49852 12.528 7.50599ZM8.372 14.704C8.18956 14.4031 8.01352 14.0984 7.844 13.79C7.749 14.047 7.661 14.3 7.581 14.551C7.838 14.607 8.102 14.658 8.371 14.704H8.372ZM10.304 14.938C11.4331 15.0219 12.5669 15.0219 13.696 14.938C14.3332 14.0019 14.9001 13.0199 15.392 12C14.9001 10.9801 14.3332 9.99804 13.696 9.06199C12.5669 8.97813 11.4331 8.97813 10.304 9.06199C9.6668 9.99804 9.09991 10.9801 8.608 12C9.09991 13.0199 9.6668 14.0019 10.304 14.938ZM16.156 10.21C16.251 9.95299 16.339 9.69999 16.419 9.44899C16.1568 9.3921 15.8934 9.34109 15.629 9.29599C15.8111 9.5969 15.9868 9.90164 16.156 10.21ZM6.13 9.83699C5.79 9.94699 5.468 10.067 5.166 10.197C3.701 10.825 3 11.535 3 12C3 12.465 3.7 13.175 5.166 13.803C5.468 13.933 5.79 14.053 6.13 14.163C6.352 13.463 6.627 12.737 6.955 12C6.63982 11.295 6.36439 10.5728 6.13 9.83699ZM7.58 9.44899C7.661 9.69899 7.749 9.95299 7.844 10.209C8.01353 9.90096 8.18957 9.59656 8.372 9.29599C8.102 9.34199 7.838 9.39299 7.581 9.44899H7.58ZM17.87 14.163C18.21 14.053 18.532 13.933 18.834 13.803C20.299 13.175 21 12.465 21 12C21 11.535 20.3 10.825 18.834 10.197C18.5183 10.0625 18.1966 9.94237 17.87 9.83699C17.648 10.537 17.373 11.263 17.045 12C17.373 12.737 17.648 13.462 17.87 14.163ZM16.42 14.551C16.339 14.301 16.251 14.047 16.156 13.791C15.9865 14.099 15.8104 14.4034 15.628 14.704C15.898 14.658 16.162 14.607 16.419 14.551H16.42Z"
      fill="currentColor"
    />
  ),
})
