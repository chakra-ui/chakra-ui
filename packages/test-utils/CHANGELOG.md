# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.6](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/test-utils@1.0.0-next.5...@chakra-ui/test-utils@1.0.0-next.6) (2020-07-15)

**Note:** Version bump only for package @chakra-ui/test-utils

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-next.5](https://github.com/chakra-ui/chakra-ui/compare/@chakra-ui/test-utils@1.0.0-next.4...@chakra-ui/test-utils@1.0.0-next.5) (2020-07-15)

**Note:** Version bump only for package @chakra-ui/test-utils

# Change Log

All notable changes to this project will be documented in this file. See
[Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.0.0-next.4 (2020-07-01)

### Bug Fixes

- ts issue with sx prop
  ([d3b1340](https://github.com/chakra-ui/chakra-ui/commit/d3b1340cb255937927b4d4c56ce218141570b951))
