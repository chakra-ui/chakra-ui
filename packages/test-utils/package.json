{
  "name": "@chakra-ui/test-utils",
  "version": "0.0.1",
  "private": true,
  "description": "Testing utilities for Chakra UI's components",
  "keywords": [
    "chakra",
    "test utils",
    "testing",
    "utils",
    "testing library",
    "react testing library"
  ],
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "homepage": "https://github.com/chakra-ui/chakra-ui#readme",
  "license": "MIT",
  "main": "src/index.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chakra-ui/chakra-ui.git"
  },
  "bugs": {
    "url": "https://github.com/chakra-ui/chakra-ui/issues"
  },
  "scripts": {
    "prebuild": "rimraf dist",
    "start": "tsdx watch",
    "build": "tsdx build",
    "test": "tsdx test --env=jsdom --passWithNoTests",
    "lint": "tsdx lint src",
    "version": "yarn build"
  },
  "devDependencies": {
    "@types/jest": "25.1.4",
    "@types/react": "16.9.23",
    "@types/react-dom": "16.9.5",
    "husky": "4.2.3",
    "react": "16.13.0",
    "react-dom": "16.13.0",
    "ts-jest": "25.2.1",
    "tsdx": "0.12.3",
    "tslib": "1.11.1",
    "typescript": "3.8.3"
  },
  "dependencies": {
    "@chakra-ui/system": "0.0.1",
    "@chakra-ui/preset-base": "0.0.1",
    "@testing-library/jest-dom": "5.1.1",
    "@testing-library/react": "9.4.0",
    "@testing-library/react-hooks": "3.2.1",
    "@testing-library/user-event": "10.0.0",
    "react-test-renderer": "16.12.0",
    "jest-axe": "3.4.0",
    "@types/jest-axe": "3.2.1"
  },
  "peerDependencies": {
    "react": "16.x",
    "react-dom": "16.x"
  }
}
