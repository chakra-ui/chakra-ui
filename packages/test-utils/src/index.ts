export * from "./test-utils"
export * from "./press"
