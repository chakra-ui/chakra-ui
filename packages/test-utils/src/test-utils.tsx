import { ChakraProvider } from "@chakra-ui/react"
import "@testing-library/jest-dom/extend-expect"
import {
  render as rtlRender,
  RenderOptions,
  fireEvent,
  RenderResult,
} from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations, axe } from "jest-axe"
import { createSerializer } from "@emotion/jest"
import { RunOptions } from "axe-core"

expect.addSnapshotSerializer(createSerializer())
expect.extend(toHaveNoViolations)

type UI = Parameters<typeof rtlRender>[0]

// UI-less passthrough fallback to prevent using conditional logic in render
function ChildrenPassthrough({ children }: { children: React.ReactElement }) {
  return children
}

export interface TestOptions extends Omit<RenderOptions, "wrapper"> {
  /**
   * optional additional wrapper, e.g. Context
   *
   * @example
   * ```ts
   * // single wrapper
   * render(<MyConponent />, {
   *  wrapper: MyContext
   * });
   *
   * // multiple wrapper
   * render(<MyConponent />, {
   *  wrapper: ({ children }) => (
   *    <ContextA>
   *      <ContextB>
   *        {children}
   *      <ContextB />
   *    <ContextA />
   *  )
   * });
   *
   * ```
   */
  wrapper?: typeof ChildrenPassthrough
}

/**
 * Custom render for @testing-library/react
 *
 * @see https://testing-library.com/docs/react-testing-library/setup#custom-render
 * @param component the component under test
 * @param options customized test options
 */
export const render = (
  ui: UI,
  { wrapper: Wrapper = ChildrenPassthrough, ...options }: TestOptions = {},
): RenderResult =>
  rtlRender(
    <ChakraProvider>
      <Wrapper>{ui}</Wrapper>
    </ChakraProvider>,
    options,
  )

export { rtlRender }
export { axe }

export * from "@testing-library/react"

export {
  act as invoke,
  renderHook,
  RenderHookOptions,
  RenderHookResult,
} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, { key: "Escape", keyCode: 27 })

type TestA11YOptions = TestOptions & { axeOptions?: RunOptions }

/**
 * Validates against common a11y mistakes.
 *
 * Wrapper for jest-axe
 *
 * @example
 * ```jsx
 * it('passes a11y test', async () => {
 *  await testA11Y(<MyComponent />, options);
 * });
 *
 * // sometimes we need to perform interactions first to render conditional UI
 * it('passes a11y test when open', async () => {
 *  const { container } = render(<MyComponent />, options);
 *
 *  fireEvent.click(screen.getByRole('button'));
 *
 *  await testA11Y(container, options);
 * });
 * ```
 *
 * @see https://github.com/nickcolley/jest-axe#testing-react-with-react-testing-library
 */
export const testA11y = async (
  ui: UI | Element,
  { axeOptions, ...options }: TestA11YOptions = {},
) => {
  const container = React.isValidElement(ui)
    ? render(ui, options).container
    : ui

  const results = await axe(container, axeOptions)

  expect(results).toHaveNoViolations()
}
