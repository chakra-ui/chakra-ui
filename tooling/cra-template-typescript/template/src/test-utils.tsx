import * as React from "react"
import { render, RenderOptions } from "@testing-library/react"
import { ChakraProvider, theme } from "@chakra-ui/react"

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ChakraProvider theme={theme}>{children}</ChakraProvider>
)

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options })

export { customRender as render }
