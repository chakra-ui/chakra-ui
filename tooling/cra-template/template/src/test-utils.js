import React from 'react';
import { render } from '@testing-library/react';
import { ChakraProvider, theme } from '@chakra-ui/react';

const AllProviders = ({ children }) => (
  <ChakraProvider theme={theme}>{children}</ChakraProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options });

export { customRender as render };
