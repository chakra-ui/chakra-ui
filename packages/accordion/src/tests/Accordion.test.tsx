import React from "react"
import { userEvent, render, fireEvent, wait } from "@chakra-ui/test-utils"
import { Accordion, AccordionButton, AccordionItem, AccordionPanel } from ".."

jest.mock("@chakra-ui/collapse", () => {
  const Collapse = jest.fn(({ children, isOpen }) => (
    <div id="collapse-mock" hidden={!isOpen}>
      {children}
    </div>
  ))
  return { Collapse }
})

test("Button renders correctly", () => {
  const { asFragment } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("uncontrolled: It opens the accordion panel", () => {
  const { getByTestId } = render(
    <Accordion defaultIndex={0}>
      <AccordionItem>
        <AccordionButton data-testid="button">Section 1 title</AccordionButton>
        <AccordionPanel data-testid="panel">Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const button = getByTestId("button")
  expect(button).toHaveAttribute("aria-expanded", "true")
})

test("uncontrolled: toggles the accordion on click", async () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Trigger</AccordionButton>
        <AccordionPanel>Panel</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const trigger = getByText("Trigger")

  userEvent.click(trigger)
  expect(trigger).toHaveAttribute("aria-expanded", "true")
  await wait()

  // you can't toggle an accordion without passing `allowToggle`
  userEvent.click(trigger)
  expect(trigger).toHaveAttribute("aria-expanded", "true")
  await wait()
})

// test that arrow up & down moves focus to next/previous accordion
test("arrow up & down moves focus to next/previous accordion", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Section 2 title</AccordionButton>
        <AccordionPanel>Panel 2</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const first = getByText("Section 1 title")
  const second = getByText("Section 2 title")

  fireEvent.keyDown(first, { key: "ArrowDown", keyCode: 40 })
  expect(second).toHaveFocus()

  fireEvent.keyDown(second, { key: "ArrowUp", keyCode: 38 })
  expect(first).toHaveFocus()
})

// test that home & end keys moves focus to first/last accordion
test("home & end keys moves focus to first/last accordion", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>First section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Second section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Last section</AccordionButton>
        <AccordionPanel>Panel 2</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const first = getByText("First section")
  const last = getByText("Last section")

  fireEvent.keyDown(first, { key: "Home", keyCode: 36 })
  expect(first).toHaveFocus()

  fireEvent.keyDown(first, { key: "End", keyCode: 35 })
  expect(last).toHaveFocus()
})

// test the only one accordion can be visible + is not togglable
test("only one accordion can be visible + is not togglable", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>First section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Second section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const firstAccordion = getByText("First section")

  userEvent.click(firstAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "true")

  userEvent.click(firstAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "true")
})
// test the only one accordion can be visible + is togglable
test("only one accordion can be visible + is togglable", () => {
  const { getByText } = render(
    <Accordion allowToggle>
      <AccordionItem>
        <AccordionButton>First section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Second section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const firstAccordion = getByText("First section")

  userEvent.click(firstAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "true")

  userEvent.click(firstAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "false")
})

// test that multiple accordions can be opened + is togglable
test("multiple accordions can be opened + is togglable", () => {
  const { getByText } = render(
    <Accordion allowMultiple>
      <AccordionItem>
        <AccordionButton>First section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Second section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const firstAccordion = getByText("First section")
  const secondAccordion = getByText("Second section")

  userEvent.click(firstAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "true")

  userEvent.click(secondAccordion)
  expect(firstAccordion).toHaveAttribute("aria-expanded", "true")
})

// it has the proper aria attributes
test("has the proper aria attributes", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const button = getByText("Section 1 title")
  const panel = getByText("Panel 1")

  expect(button).toHaveAttribute("aria-controls")
  expect(button).toHaveAttribute("aria-expanded")
  expect(panel).toHaveAttribute("aria-labelledby")
})

// test that tab moves focus to the next focusable element
test("tab moves focus to the next focusable element", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>First section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Second section</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>

      <AccordionItem>
        <AccordionButton>Last section</AccordionButton>
        <AccordionPanel>Panel 2</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const first = getByText("First section")
  const second = getByText("Second section")
  const last = getByText("Last section")

  userEvent.tab()
  expect(first).toHaveFocus()

  userEvent.tab()
  expect(second).toHaveFocus()

  userEvent.tab()
  expect(last).toHaveFocus()
})

// test that aria-contols for button is same as id for panel
test("aria-contols for button is same as id for panel", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const button = getByText("Section 1 title")
  const panel = getByText("Panel 1")
  expect(button.getAttribute("aria-controls")).toEqual(panel.getAttribute("id"))
})

// test that aria-expanded is true/false when accordion is open/closed
test("aria-expanded is true/false when accordion is open/closed", () => {
  const { getByText } = render(
    <Accordion defaultIndex={0}>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
      <AccordionItem>
        <AccordionButton>Section 2 title</AccordionButton>
        <AccordionPanel>Panel 2</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )

  const button = getByText("Section 1 title")
  expect(button).toHaveAttribute("aria-expanded", "true")
})

// test that panel has role=region and aria-labelledby
test("panel has role=region and aria-labelledby", () => {
  const { getByText } = render(
    <Accordion>
      <AccordionItem>
        <AccordionButton>Section 1 title</AccordionButton>
        <AccordionPanel>Panel 1</AccordionPanel>
      </AccordionItem>
    </Accordion>,
  )
  const panel = getByText("Panel 1")

  expect(panel).toHaveAttribute("aria-labelledby")
  expect(panel).toHaveAttribute("role", "region")
})
