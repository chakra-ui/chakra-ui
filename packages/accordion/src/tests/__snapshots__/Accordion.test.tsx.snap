// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <div
    data-chakra-accordion=""
  >
    <div
      data-chakra-accordion-item=""
    >
      <button
        aria-controls="accordion-panel-1"
        aria-expanded="false"
        class="css-so65e"
        data-chakra-accordion-button=""
        id="accordion-button-1"
      >
        Section 1 title
      </button>
      <div
        hidden=""
        id="collapse-mock"
      >
        <div
          aria-labelledby="accordion-button-1"
          class="css-t2ecxu"
          data-chakra-accordion-panel=""
          id="accordion-panel-1"
          role="region"
        >
          Panel 1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
